libbeat/cmd/instance/beat.go

```diff
@@ -999,3 +996,15 @@ func obfuscateConfigOpts() []ucfg.Option {
 		ucfg.ResolveNOOP,
 	}
 }
+
+// LoadKeystore returns the appropriate keystore based on the configuration.
+func LoadKeystore(cfg *common.Config, name string) (keystore.Keystore, error) {
+	keystoreCfg, _ := cfg.Child("keystore", -1)
+	defaultPathConfig, _ := cfg.String("path.config", -1)
+	defaultPathConfig = filepath.Join(defaultPathConfig, fmt.Sprintf("%s.keystore", name))
+	store, err := keystore.Factory(keystoreCfg, defaultPathConfig)
+	if err != nil {
+		return nil, err
+	}
+	return store, nil
+}
```

Comment:
we should not ignore errors here.

The config path is better read via `paths` package. This one is also aware of CLI overwrites. The paths package is initialized during config loading.

The keystore config will be nil if there was an error. Do we want to pass a `nil` to the factor in this case?

the if condition can be removed:

```
return callThatCanFail()
```

should be equivalent to:

```
ret, err := callThatCanFail()
if err != nil {
  return nil, err
}
return ret, nil
```

callers should never use ret without checking the error when a function returns an interface value.